test/index.test.ts

```diff
@@ -208,6 +209,103 @@ describe('PaymentSocket', function () {
     })
   })
 
+  describe('pay', function () {
+    it('should lower the maximum balance and resolve when the balance is below the maximum', async function () {
+      await this.clientSocket.pay(2000)
+      assert.equal(this.clientSocket.balance, '-2000')
+
+      await this.clientSocket.pay(1500)
+      assert.equal(this.clientSocket.balance, '-3500')
+    })
+
+    it('should reject if the amount is higher than the other party will accept', async function () {
+      this.serverSocket.setMaxBalance(900)
+      let errored = false
+      try {
+        await this.clientSocket.pay(2000)
```

Comment:
Does this throw before sending anything?

Reply:
Not yet but there's an issue open about that

Reply:
See https://github.com/emschwartz/ilp-protocol-paystream/issues/11